app/services/foreman_templates/template_importer.rb

```diff
@@ -79,11 +78,32 @@ def parse_files!
     end
 
     def import_options
-      { :force => @force,
+      lock_predicate = lambda do |template|
+        case @lock
+        when 'lock'
+          return true
+        when 'unlock'
+          return false
+        when 'keep'
+          return template.locked unless template.new_record?
+
+          return false
```

Comment:
Would it do the same if we have `template.new_record? ? false : template.locked` ?